pjsip/src/pjsip/sip_transport_tcp.c

```diff
@@ -1507,6 +1550,20 @@ static pj_bool_t on_connect_complete(pj_activesock_t *asock,
     return PJ_TRUE;
 }
 
+/* Transport connection timeout timer callback */
```

Comment:
I think this is safe because

*   `on_connect_complete` itself never destroys the `pjsip_transport` structure - it just calls `pjsip_transport_dec_ref`
*   `pjsip_transport_dec_ref` sets a timer to destroy the `pjsip_transport` structure
*   when this timer pops, `tcp_destroy` is called, which cancels the connect timer
*   all timers are handled by a single thread, so the connect timer can't pop while the destruction timer is popping.

I think this argument applies equally to the `ka_timer`.

Is that OK?